### PR TITLE
reducing generated flow types

### PR DESCRIPTION
**Pull request by chrisnojima**
During a conversation w/ max he mentioned all the avdl based enums/records/etc are uniquely named (for go) . we were prefixing the protocol name to avoid collisions but we can just remove all that now. This collapses a bunch of types (once avro handles the avdl we lose the fact that it imported stuff so we couldn't glean that common.avdl exists, etc)

tldr; less flow type duplication

@keybase/react-hackers low priority

### CODE REVIEW

File: go/protocol/keybase_v1.go [hunk 1]
```diff
@@ -217,6 +217,14 @@ type UserPlusKeys struct {
 	Uvv        UserVersionVector `codec:"uvv" json:"uvv"`
 }
 
+type MerkleTreeID int
```
Comment (chrisnojima) [inline go/protocol/keybase_v1.go]: So this got added but protocol wasn't ever genned / checked in after so thats why this exists

File: protocol/bin/flow.js [hunk 2]
```diff
@@ -1,20 +1,19 @@
-'use strict'
+'use strict' // eslint-disable-line
```
Comment (chrisnojima) [inline protocol/bin/flow.js]: just some general cleanup of this file. more arrows and es6 stuff that node handles out of the box

File: protocol/bin/flow.js [hunk 3]
```diff
@@ -69,32 +68,32 @@ function figureType (protocol, rawType) {
...
-    const responseType = figureType(json.protocol, message.response)
-    let response = '/* void response */'
-    if (responseType !== 'null') {
-      response = `export type ${name}_result = ${responseType}`
-    }
+    const responseType = figureType(message.response)
+    const response = `export type ${name}_result = ${responseType === 'null' ? 'void' : responseType}`
 
     const isNotify = message.hasOwnProperty('notify')
-    let r = ' /* Notify call, No response */'
```
Comment (chrisnojima) [inline protocol/bin/flow.js]: instead of a comment we generate the void type

File: protocol/bin/flow.js [hunk 4]
```diff
@@ -69,32 +68,32 @@ function figureType (protocol, rawType) {
...
+    let r = null
     if (!isNotify) {
       const type = (responseType === 'null') ? '' : `result: ${name}_result`
       r = `,\n    response: {
       error: (err: string) => void,
       result: (${type}) => void
     }`
+    } else {
+      r = ` /* ,\n    response: {} // Notify call
```
Comment (chrisnojima) [inline protocol/bin/flow.js]: let's keep the same structure, but comment it out

File: protocol/bin/flow.js [hunk 5]
```diff
@@ -119,43 +118,10 @@ function analyzeMessages (json) {
   callback: (null | (err: ?any${r}) => void)
 }`
 
-    return [
-      `// ${json.protocol}.${m} ////////////////////////////////////////`,
-      response, rpc, ''].join('\n\n')
```
Comment (chrisnojima) [inline protocol/bin/flow.js]: we're sorting all the things now so splitting up the sections doesn't make sense

File: protocol/bin/flow.js [hunk 6]
```diff
@@ -197,7 +163,7 @@ function parseMaybe (t) {
 }
 
 function parseUnion (unionTypes) {
-  return unionTypes.map(parseInnerType).join(' | ')
+  return '\n    ' + unionTypes.map(parseInnerType).join('\n  | ')
```
Comment (MarcoPolo) [inline protocol/bin/flow.js]: Does this work that way? maybe we should use the new format with a leading `|` https://github.com/facebook/flow/blob/master/Changelog.md

**Comment by MarcoPolo** [comment]
Does this change the flow type names we're importing currently?

**Comment by chrisnojima** [comment]
the rpcs and responses should be the same. the flow type still checks fine (seemingly)

File: protocol/bin/flow.js (hunk 6 above)
Comment (chrisnojima) [inline protocol/bin/flow.js]: ? this does generate that

**Comment by chrisnojima** [comment]
updated to flow 22 and found one change, now fixed. @MarcoPolo

File: protocol/bin/flow.js (hunk 6 above)
Comment (MarcoPolo) [inline protocol/bin/flow.js]: this generates:
```
type Foo =
     one
  |  two
  | three
```
instead of 
```
type Foo =
  |  one
  |  two
  | three
```

**Comment by MarcoPolo** [comment]
:+1:

File: protocol/bin/flow.js (hunk 6 above)
Comment (chrisnojima) [inline protocol/bin/flow.js]: ah... meh, i kinda don't like that they support this kinda weird syntax

Comment (MarcoPolo) [inline protocol/bin/flow.js]: If the other one works that's fine too!

**Pull request merged**